smart-contract/contracts/CustomToken.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract CustomToken is Ownable{
+
+    struct Token {
+        string name;
+        string symbol;
+        string tokenType;
+        uint initialSupply;
+        uint decimals;
+        bool isBurnable;
+        bool isMintable;
+        bool doesHolderRedistribution;
+        bool hasFeesOrTaxes;
+        uint8 holdersRewardFee;
+        uint8 burnFeePercentage;
+        uint8 transactionTaxPercentage;
+        address taxReceivingAddress;
+    }
+
+    event Created(string _name, string _symbol, string _tokenType);
+    event Returned(address owner);
+
+    mapping(address => Token[]) createdTokens;
+
+    modifier validString(string memory _string)
+    {
+        require(!isNullOrWhitespace(_string), "String is null or whitespace");
+        _;
+    }
+
+    function isNullOrWhitespace(string memory _content) pure private returns(bool)
+    {
+        require(bytes(_content).length > 0, "String is null or empty");
+
+        if (bytes(_content).length < 3) { return true; }
+
+        bytes memory contentBytes = bytes(_content);
+
+        for (uint8 _index = 2; _index < contentBytes.length; _index++) {
```

Comment:
can you add a comment on what this for loop is doing? Why is it indexing from 2 and %2?